src/App/Tracker/Controller/Hooks/Listeners/AbstractListener.php

```diff
@@ -0,0 +1,210 @@
+<?php
+/**
+ * Part of the Joomla Tracker's Tracker Application
+ *
+ * @copyright  Copyright (C) 2012 - 2015 Open Source Matters, Inc. All rights reserved.
+ * @license    http://www.gnu.org/licenses/gpl-2.0.txt GNU General Public License Version 2 or Later
+ */
+
+namespace App\Tracker\Controller\Hooks\Listeners;
+
+use App\Tracker\Table\IssuesTable;
+use Joomla\Event\Event;
+use Joomla\Github\Github;
+
+use Monolog\Logger;
+
+/**
+ * Abstract listener class for custom Listeners
+ *
+ * @since  1.0
+ */
+abstract class AbstractListener
+{
+	/**
+	 * Check if label already exists
+	 *
+	 * @param   object  $hookData    Hook data payload
+	 * @param   Github  $github      Github object
+	 * @param   Logger  $logger      Logger object
+	 * @param   object  $project     Object containing project data
+	 * @param   string  $checkLabel  The label to check
+	 *
+	 * @return  bool    True if the label already exists
+	 *
+	 * @since   1.0
+	 */
+	protected function checkLabel($hookData, Github $github, Logger $logger, $project, $checkLabel)
+	{
+		// The Github ID if we have a pull or issue so that method can handle both
+		$issueNumber = $this->getIssueID($hookData);
+
+		// Get the labels for the pull's issue
+		try
+		{
+			$labels = $github->issues->get($project->gh_user, $project->gh_project, $issueNumber)->labels;
+		}
+		catch (\DomainException $e)
+		{
+			$logger->error(
+				sprintf(
+					'Error retrieving labels for GitHub item %s/%s #%d - %s',
+					$project->gh_user,
+					$project->gh_project,
+					$issueNumber,
+					$e->getMessage()
+				)
+			);
+
+			return false;
```

Comment:
Let's change this line to `throw new RuntimeException;`. Otherwise this exception will mean that label is not found as we are returning `false` here. But it is not correct, we have an exception.